src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0
+      || this.b <= 0
+      || this.c <= 0) {
+      throw new Error('Invalid value.');
+    }
 
-export class Circle {
+    if (this.a + this.b <= this.c
+      || this.a + this.c <= this.b
+      || this.b + this.c <= this.a) {
+      throw new Error('Invalid sizes length.');
+    }
+  }
 
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) / 2;
+
+    return +Math.sqrt(
+      p * (p - this.a) * (p - this.b) * (p - this.c),
+    ).toFixed(2);
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (this.radius < 0) {
```

Comment:
```suggestion
    if (this.radius <= 0) {
```
The constructors should `throw new Error('your error message')` if any length is <= 0